lib/lintPostcssResult.mjs

```diff
@@ -154,3 +157,23 @@ function warnDeprecatedRule(result, ruleName) {
 
 	result.warn(message, { stylelintType: 'deprecation' });
 }
+
+/** @type {import('css-tree').Lexer | null} */
+let cachedLexer = null;
+
+/**
+ * @param {Config} config
+ * @returns {import('css-tree').Lexer}
+ * */
+function getCachedLexer(config) {
+	if (!cachedLexer) {
+		cachedLexer = fork({
+			atrules: config.languageOptions?.syntax?.atRules || {},
+			properties: config.languageOptions?.syntax?.properties || {},
+			types: config.languageOptions?.syntax?.types || {},
+			cssWideKeywords: config.languageOptions?.syntax?.cssWideKeywords || [],
+		}).lexer;
+	}
+
+	return cachedLexer;
+}
```

Comment:
This now uses a "per process" cache.

In practice this might lead to unexpected behavior because a single process might lint multiple files, each with different config and different values for `languageOptions`.

Instead we should cache per call to `lintPostcssResult`.

This still has the benefit of only creating one lexer fork for all rules for each lint.

Reply:
Does this mean that if `a.css` and `b.css` have different configs, the first `lexer` could be reused for both and cause issues? I wasn't sure how common it is for a single process to lint files with different configs, but it seems like the possibility isn't zero.

Your suggestion to move the cache to the function level makes sense to avoid this. 
I'll implement that change. Thanks for the insightful feedback!

Reply:
I changed the cache to be managed using the `syntax` as the key. What do you think?
[Change lexer cache to use syntax-based key to prevent cache pollution](https://github.com/stylelint/stylelint/pull/8297/commits/ee23855da4906c15e9f8b7e3ffdc2878c04aec79)

Reply:
@romainmenke 
Can you see if there is any problem with this change?